### PR TITLE
example: add create namespace kubectl equivalent

### PR DESCRIPTION
**Pull request by Ananya2001-an**
## Description

Related to https://github.com/fabric8io/kubernetes-client/issues/4267

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by Ananya2001-an** [comment]
In my local maven project I am able to create a namespace but every time I try and execute the file it also gives me some abrupt thread error....I am not sure what's the issue here?
Screenshot 2023-09-06 004202

**Comment by Ananya2001-an** [comment]
> @Ananya2001-an : Which issue is this PR trying to fix?

This was an old issue that was closed but was yet to complete here #4267

**Comment by rohanKanojia** [comment]
@Ananya2001-an : I've updated the PR description. You should try to provide as much information about the changes you've made in the PR description so that reviewers can get better context

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      logger.info("Kubernetes client successfully created");
+
+      // Create a namespace
+      Namespace namespace = new NamespaceBuilder()
+          .withNewMetadata()
+          .withName("test")
+          .endMetadata()
+          .build();
+      logger.info("Created namespace: {}", client.namespaces().resource(namespace).create().getMetadata().getName());
+    } catch (KubernetesClientException aException) {
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java:44]:
```suggestion
    } catch (KubernetesClientException exception) {
```

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      logger.error("Problem encountered in Kubernetes Client");
+      aException.printStackTrace();
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java:46]:
```suggestion
      exception.printStackTrace();
```

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    try (final KubernetesClient client = new KubernetesClientBuilder().build()) {
+      logger.info("Kubernetes client successfully created");
+
+      // Create a namespace
+      Namespace namespace = new NamespaceBuilder()
+          .withNewMetadata()
+          .withName("test")
+          .endMetadata()
+          .build();
+      logger.info("Created namespace: {}", client.namespaces().resource(namespace).create().getMetadata().getName());
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java:43]: I would separate namespace creation and logging to separate lines:

```java
namespace = client.namespaces().resource(namespace).create();
logger.info("Created namespace: {}", namespace.getMetadata().getName());
```

**Comment by Ananya2001-an** [comment]
> In my local maven project I am able to create a namespace but every time I try and execute the file it also gives me some abrupt thread error....I am not sure what's the issue here? Screenshot 2023-09-06 004202

Can u tell me something about this?

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      Namespace namespace = new NamespaceBuilder()
+          .withNewMetadata()
+          .withName("test")
+          .endMetadata()
+          .build();
+      namespace = client.namespaces().resource(namespace).create();
+      logger.info("Created namespace: {}", namespace.getMetadata().getName());
+    } catch (KubernetesClientException exception) {
+      logger.error("Problem encountered in Kubernetes Client");
+      exception.printStackTrace();
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java:47]: Sonar seems to be having problems with this. Can we just add this to logger ?
```
logger.error("Problem encountered in Kubernetes Client: {}", exception.getMessage());
```

Comment (Ananya2001-an) [reply kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/NamespaceCreateEquivalent.java:47]: yeah sure

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@Ananya2001-an : Could you please squash your commits into one?

> can u also tell me something about this?

Is it possible to share complete stacktrace? Not 100% sure but it might be underlying okhttp client hasn't closed ...

**Comment by Ananya2001-an** [comment]
> Is it possible to share complete stacktrace? Not 100% sure but it might be underlying okhttp client hasn't closed ...

But if I am using the try-with-resources block, it should have automatically closed the http client, right?

Screenshot 2023-09-07 151029

**Comment by Ananya2001-an** [comment]
@rohanKanojia I am not sure why the build is failing....can u look into it?

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**